CMakeLists.txt

```diff
@@ -592,80 +656,106 @@ elseif("${SWIFT_HOST_VARIANT_SDK}" MATCHES "(OSX|IOS*|TVOS*|WATCHOS*)")
   message(STATUS "${xcode_version}")
   message(STATUS "")
 
-  is_sdk_requested(OSX swift_build_osx)
-  if(swift_build_osx)
-    configure_sdk_darwin(
-        OSX "OS X" "${SWIFT_DARWIN_DEPLOYMENT_VERSION_OSX}"
-        macosx macosx macosx "x86_64")
+  # Compatible host/cross-compile SDKS for Darwin OSes:
+  # OSX, IOS(_SIMULATOR), TVOS(_SIMULATOR), WATCHOS(_SIMULATOR)
+
+  # - configure_sdk_darwin handles the architectures for each SDK,
+  #   so we need to gather them up (including merging host/cross targets in one list)
+  # - configure_sdk_darwin calculates its own triple, so
+  #   Darwin target validation happens here.
+
+  set(OSX_VALID_ARCHITECTURES "x86_64")
+  set(IOS_VALID_ARCHITECTURES "armv7;armv7s;arm64")
+  set(TVOS_VALID_ARCHITECTURES "arm64")
+  set(WATCHOS_VALID_ARCHITECTURES "armv7k")
+  set(IOS_SIMULATOR_VALID_ARCHITECTURES "i386;x86_64")
+  set(TVOS_SIMULATOR_VALID_ARCHITECTURES "x86_64")
+  set(WATCHOS_SIMULATOR_VALID_ARCHITECTURES "i386")
+
+  foreach(stdlib_target ${SWIFT_CROSS_COMPILE_STDLIB_TARGETS})
+    get_sdk_from_stdlib_target("${stdlib_target}" target_sdk)
+    get_arch_from_stdlib_target("${stdlib_target}" target_arch)
+
+    if(NOT "${target_arch}" IN_LIST ${target_sdk}_VALID_ARCHITECTURES)
```

Comment:
Looks like `IN_LIST` is new in CMake 3.3. (https://cmake.org/cmake/help/v3.3/policy/CMP0057.html)

We support Ubuntu 14.04, which has CMake 2.8.12.